src/ingestion/process-event.ts

```diff
@@ -1,38 +1,455 @@
+            if (otherPerson.created_at < firstSeen) {
+                // Keep the oldest created_at (i.e. the first time we've seen this person)
+                firstSeen = otherPerson.created_at
+            }
+        }
+
+        await this.db.updatePerson(mergeInto, { created_at: firstSeen })
+
+        // merge the distinct_ids
+        for (const otherPerson of peopleToMerge) {
+            const otherPersonDistinctIds: PersonDistinctId[] = (
+                await this.db.postgresQuery(
+                    'SELECT * FROM posthog_persondistinctid WHERE person_id = $1 AND team_id = $2',
+                    [otherPerson, mergeInto.team_id]
+                )
+            ).rows
+            for (const personDistinctId of otherPersonDistinctIds) {
+                await this.db.updateDistinctId(personDistinctId, { person_id: mergeInto.id })
+            }
+
+            const otherPersonCohortIds: CohortPeople[] = (
+                await this.db.postgresQuery('SELECT * FROM posthog_cohortpeople WHERE person_id = $1', [otherPerson.id])
+            ).rows
+            for (const personCohortId of otherPersonCohortIds) {
+                await this.db.postgresQuery('UPDATE posthog_cohortpeople SET person_id = $1 WHERE id = $2', [
+                    mergeInto.id,
+                    personCohortId.id,
+                ])
+            }
+
+            await this.db.deletePerson(otherPerson.id)
+        }
+    }
+
+    private async captureEE(
+        eventUuid: string,
+        personUuid: string,
+        ip: string,
+        siteUrl: string,
+        teamId: number,
+        event: string,
+        distinctId: string,
+        properties: Properties,
+        timestamp: DateTime,
+        sentAt: DateTime | null
+    ): Promise<IEvent> {
+        event = sanitizeEventName(event)
+
+        const elements: Record<string, any>[] | undefined = properties['$elements']
+        let elements_list: Element[] = []
+        if (elements && elements.length) {
+            delete properties['$elements']
+            elements_list = elements.map((el) => ({
+                text: el['$el_text'] ? el['$el_text'].slice(0, 400) : null,
+                tag_name: el['tag_name'],
+                href: el['attr__href'] ? el['attr__href'].slice(0, 2048) : null,
+                attr_class: el['attr__class'] ? el['attr__class'].split(' ') : null,
+                attr_id: el['attr__id'],
+                nth_child: el['nth_child'],
+                nth_of_type: el['nth_of_type'],
+                attributes: Object.fromEntries(Object.entries(el).filter(([key]) => key.startsWith('attr__'))),
+            }))
+        }
+
+        const team: Team = (await this.db.postgresQuery('SELECT * FROM posthog_team WHERE id = $1', [teamId])).rows[0]
+
+        if (!team.anonymize_ips && !('$ip' in properties)) {
+            properties['$ip'] = ip
+        }
+
+        this.storeNamesAndProperties(team, event, properties)
```

Comment:
Yeah, I saw that but this was the only thing I held off on porting because of the complexity. Per our discussion, now handled earlier in the pipeline, in the capture endpoint